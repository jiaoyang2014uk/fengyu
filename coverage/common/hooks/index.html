
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for common/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> common/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>266/266</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">96.42% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>54/56</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>18/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>266/266</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="useClass.ts"><a href="useClass.ts.html">useClass.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	</tr>

<tr>
	<td class="file high" data-value="useColumns.ts"><a href="useColumns.ts.html">useColumns.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	</tr>

<tr>
	<td class="file high" data-value="useFormattedData.ts"><a href="useFormattedData.ts.html">useFormattedData.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="105" class="abs high">105/105</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="105" class="abs high">105/105</td>
	</tr>

<tr>
	<td class="file high" data-value="usePagination.ts"><a href="usePagination.ts.html">usePagination.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="37" class="abs high">37/37</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="37" class="abs high">37/37</td>
	</tr>

<tr>
	<td class="file high" data-value="useSortData.ts"><a href="useSortData.ts.html">useSortData.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="31" class="abs high">31/31</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="31" class="abs high">31/31</td>
	</tr>

<tr>
	<td class="file high" data-value="useSortable.ts"><a href="useSortable.ts.html">useSortable.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Wed May 25 2022 00:59:48 GMT+0800 (China Standard Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    