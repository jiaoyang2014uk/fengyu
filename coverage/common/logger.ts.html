
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for common/logger.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> / <a href="index.html">common</a> logger.ts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">87.87% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>87/99</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.76% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>21/26</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">82.35% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>14/17</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">87.87% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>87/99</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">309x</span>
<span class="cline-any cline-yes">309x</span>
<span class="cline-any cline-yes">309x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">309x</span>
<span class="cline-any cline-yes">289x</span>
<span class="cline-any cline-yes">289x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">309x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">215x</span>
<span class="cline-any cline-yes">215x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">74x</span>
<span class="cline-any cline-yes">74x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/**
 * 根据环境不同设置日志级别
 * 使用方法：
 * 设置级别：setLogLevel(levelTypeName)，由高到低分别为：trace，debug，info，warn，error， none
 * 调用日志方法：log(levelTypeName, ...args)或者形如trace(...args)
 */
const isDev = process.env.NODE_ENV === 'development';
&nbsp;
export enum <span class="fstat-no" title="function not covered" >LEVEL_TYPE {</span>
    trace = 'trace',
    debug = 'debug',
    info = 'info',
    warn = 'warn',
    error = 'error',
    none = 'none',
}
&nbsp;
const levels: Record&lt;string, string[]&gt; = {
    [LEVEL_TYPE.trace]: [LEVEL_TYPE.trace, LEVEL_TYPE.debug, LEVEL_TYPE.info, LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.debug]: [LEVEL_TYPE.debug, LEVEL_TYPE.info, LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.info]: [LEVEL_TYPE.info, LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.warn]: [LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.error]: [LEVEL_TYPE.error],
    [LEVEL_TYPE.none]: []
}
&nbsp;
// 生成日志头信息
function getLogInfo() {
    const date = new Date();
    return `[${date.toLocaleDateString()}-${date.toLocaleTimeString()}]: `;
}
&nbsp;
let openLevels: string[] = levels[LEVEL_TYPE.info]
&nbsp;
function isOpen(levelType = LEVEL_TYPE.info) {
    return openLevels.includes(levelType)
}
&nbsp;
export function log(levelType = LEVEL_TYPE.info, ...args: unknown[]) {
    if (!isOpen(levelType)) {
        return
    }
    const logInfo = getLogInfo()
    switch (<span class="branch-0 cbranch-no" title="branch not covered" >levelType) {</span>
<span class="cstat-no" title="statement not covered" >        case "trace":</span>
<span class="cstat-no" title="statement not covered" >            console.trace(logInfo, ...args);</span>
<span class="cstat-no" title="statement not covered" ><span class="branch-0 cbranch-no" title="branch not covered" >            break;</span></span>
<span class="cstat-no" title="statement not covered" >        case "debug":</span>
<span class="cstat-no" title="statement not covered" >            console.debug(logInfo, ...args);</span>
<span class="cstat-no" title="statement not covered" >            break;</span>
        case "info":
            console.info(logInfo, ...args);
            break;
        case "warn":
            console.warn(logInfo, ...args);
            break;
        case "error":
            console.error(logInfo, ...args);
<span class="branch-0 cbranch-no" title="branch not covered" >            break;</span>
<span class="cstat-no" title="statement not covered" >        default:</span>
<span class="cstat-no" title="statement not covered" >            console.info(logInfo, ...args);</span>
<span class="cstat-no" title="statement not covered" >            break;</span>
    }
<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >}</span></span>
&nbsp;
// 实时调试日志
export function trace(...args: unknown[]) {
    log(LEVEL_TYPE.trace, ...args)
}
&nbsp;
// 调试日志
export function debug(...args: unknown[]) {
    log(LEVEL_TYPE.debug, ...args)
}
&nbsp;
// 信息日志
export function info(...args: unknown[]) {
    log(LEVEL_TYPE.info, ...args)
}
&nbsp;
// 告警日志
export function warn(...args: unknown[]) {
    log(LEVEL_TYPE.warn, ...args)
}
&nbsp;
// 错误日志
export function error(...args: unknown[]) {
    log(LEVEL_TYPE.error, ...args)
}
&nbsp;
export function setLogLevel(level = LEVEL_TYPE.info) {
    openLevels = levels[<span class="branch-0 cbranch-no" title="branch not covered" >level] || []</span>
<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >}</span></span>
&nbsp;
if (isDev) <span class="branch-0 cbranch-no" title="branch not covered" >{</span>
<span class="cstat-no" title="statement not covered" >    setLogLevel(LEVEL_TYPE.trace)</span>
} else {
    setLogLevel(LEVEL_TYPE.info)
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Wed May 25 2022 00:59:48 GMT+0800 (China Standard Time)
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    