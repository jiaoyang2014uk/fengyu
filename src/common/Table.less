//table
.table {
  &__table {
    width: 100%;
    border-collapse: collapse;
  }

  &__body-cell,
  &__header-cell {
    border: 1px solid #ddd;
    padding: 8px;
  }

  &__header-cell {
    background-color: #f1f4f8;
    cursor: pointer;

    &_sort,
    &_sort_asc,
    &_sort_desc {
      &::after {
        margin-left: 8px;
        content: '↑↓';
      }

      &_desc::after,
      &_asc::after {
        color: red;
      }

      &_desc::after {
        content: '↓';
      }

      &_asc::after {
        content: '↑';
      }
    }
  }
}

//pagination
.pagination {
  text-align: right;
  margin-top: 12px;

  &__previous,
  &__next {
    color: blue;
    cursor: pointer;
    margin-left: 12px;

    &.disabled {
      cursor: unset;
      color: grey;
    }
  }

  &__page,
  &__goto {
    margin-left: 12px;

    &-input {
      width: 30px;
      margin-left: 4px;
      border: 1px solid #999;
    }
  }
}

