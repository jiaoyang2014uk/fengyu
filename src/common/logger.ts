/**
 * 根据环境不同设置日志级别
 * 使用方法：
 * 设置级别：setLogLevel(levelTypeName)，由高到低分别为：trace，debug，info，warn，error， none
 * 调用日志方法：log(levelTypeName, ...args)或者形如trace(...args)
 */
const isDev = process.env.NODE_ENV === 'development';

export enum LEVEL_TYPE {
    trace = 'trace',
    debug = 'debug',
    info = 'info',
    warn = 'warn',
    error = 'error',
    none = 'none',
}

const levels: Record<string, string[]> = {
    [LEVEL_TYPE.trace]: [LEVEL_TYPE.trace, LEVEL_TYPE.debug, LEVEL_TYPE.info, LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.debug]: [LEVEL_TYPE.debug, LEVEL_TYPE.info, LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.info]: [LEVEL_TYPE.info, LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.warn]: [LEVEL_TYPE.warn, LEVEL_TYPE.error],
    [LEVEL_TYPE.error]: [LEVEL_TYPE.error],
    [LEVEL_TYPE.none]: []
}

// 生成日志头信息
function getLogInfo() {
    const date = new Date();
    return `[${date.toLocaleDateString()}-${date.toLocaleTimeString()}]: `;
}

let openLevels: string[] = levels[LEVEL_TYPE.info]

function isOpen(levelType = LEVEL_TYPE.info) {
    return openLevels.includes(levelType)
}

export function log(levelType = LEVEL_TYPE.info, ...args: unknown[]) {
    if (!isOpen(levelType)) {
        return
    }
    const logInfo = getLogInfo()
    switch (levelType) {
        case "trace":
            console.trace(logInfo, ...args);
            break;
        case "debug":
            console.debug(logInfo, ...args);
            break;
        case "info":
            console.info(logInfo, ...args);
            break;
        case "warn":
            console.warn(logInfo, ...args);
            break;
        case "error":
            console.error(logInfo, ...args);
            break;
        default:
            break;
    }
}

// 实时调试日志
export function trace(...args: unknown[]) {
    log(LEVEL_TYPE.trace, ...args)
}

// 调试日志
export function debug(...args: unknown[]) {
    log(LEVEL_TYPE.debug, ...args)
}

// 信息日志
export function info(...args: unknown[]) {
    log(LEVEL_TYPE.info, ...args)
}

// 告警日志
export function warn(...args: unknown[]) {
    log(LEVEL_TYPE.warn, ...args)
}

// 错误日志
export function error(...args: unknown[]) {
    log(LEVEL_TYPE.error, ...args)
}

function setLogLevel(level = LEVEL_TYPE.info) {
    openLevels = levels[level]
}

if (isDev) {
    setLogLevel(LEVEL_TYPE.trace)
} else {
    setLogLevel(LEVEL_TYPE.info)
}
