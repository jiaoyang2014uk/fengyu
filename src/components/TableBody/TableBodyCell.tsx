import {defineComponent, computed} from "vue";
import type {TableBodyCellProps} from "../../common/types";
import {tableBodyCellProps} from "../../common/const";

export default defineComponent({
    name: "TableBodyCell",
    props: tableBodyCellProps,
    setup(props: TableBodyCellProps) {
        const value = computed(() => props.data[props.column.dataIndex]);
        return () => {
            return (
                <td class="table__body-cell">
                    <span>{props.column.slot ? props.column.slot({
                        value: value.value,
                        data: props.data
                    }) : value.value}</span>
                </td>
            );
        };
    },
});

