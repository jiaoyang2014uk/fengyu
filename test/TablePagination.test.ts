import {expect, test} from "vitest";
import {mountTable, propsPartial} from './common'
import cloneDeep from 'lodash/cloneDeep';
import TableBodyCell from "../src/components/TableBody/TableBodyCell";

test('pagination: pageSize > data.length', () => {
    const wrapper = mountTable({
        props: {
            ...cloneDeep(propsPartial),
            paginationOptions: {
                pageSize: 5,
                current: 1
            }
        },
    });

    expect(wrapper.html()).toMatchSnapshot();
});

test('pagination: pageSize < data.length', async () => {
    const wrapper = mountTable({
        props: {
            ...cloneDeep(propsPartial),
            paginationOptions: {
                pageSize: 2,
                current: 1
            }
        },
    });

    //共2页，当前第1页
    expect(wrapper.find('.pagination__page').html()).toContain('( 1 / 2 )');

    // 在第一页点击上一页，无效果
    await wrapper.findAll('.pagination__previous')[0].trigger('click')
    expect(wrapper.findAllComponents(TableBodyCell)[0].html()).toContain('Bob');

    // 点击下一页
    await wrapper.findAll('.pagination__next')[0].trigger('click')
    expect(wrapper.findAllComponents(TableBodyCell)[0].html()).not.toContain('Bob');
    expect(wrapper.findAllComponents(TableBodyCell)[0].html()).toContain('Alice');


    // 在最后一页点击下一页，无效果
    await wrapper.findAll('.pagination__next')[0].trigger('click')
    expect(wrapper.findAllComponents(TableBodyCell)[0].html()).toContain('Alice');

    // 跳页到第一页
    wrapper.findAll('.pagination__goto-input')[0].setValue(1)
    await wrapper.findAll('.pagination__goto-input')[0].trigger('blur')
    expect(wrapper.findAllComponents(TableBodyCell)[0].html()).toContain('Bob');
});

test('pagination: remote', async () => {
    const wrapper = mountTable({
        props: {
            ...propsPartial,
            paginationOptions: {
                pageSize: 2,
                current: 1,
                remote: true
            }
        },
    });


    // 点击下一页，无效果
    await wrapper.findAll('.pagination__next')[0].trigger('click')
    expect(wrapper.findAllComponents(TableBodyCell)[0].html()).toContain('Bob');

});

