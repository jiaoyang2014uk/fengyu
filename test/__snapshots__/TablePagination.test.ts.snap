// Vitest Snapshot v1

exports[`pagination: pageSize > data.length 1`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <colgroup>
      <col width=\\"200\\">
      <col>
    </colgroup>
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <div class=\\"pagination\\"><span class=\\"pagination__total\\">Total 3 items</span><span class=\\"pagination__previous disabled\\">Previous</span><span class=\\"pagination__next disabled\\">Next</span><span class=\\"pagination__page\\">( 1 / 1 )</span><span class=\\"pagination__goto\\">Go to<input type=\\"text\\" class=\\"pagination__goto-input\\"></span></div>
</div>"
`;
